lib/chewy/type/syncer.rb

```diff
@@ -56,7 +56,8 @@ class Syncer
       def self.typecast_date(string)
         if string.is_a?(String) && (match = ISO_DATETIME.match(string))
           microsec = (match[7].to_r * 1_000_000).to_i
-          date = "#{match[1]}-#{match[2]}-#{match[3]}T#{match[4]}:#{match[5]}:#{match[6]}.#{format('%06d', microsec)}+00:00"
+          date = "#{match[1]}-#{match[2]}-#{match[3]}T#{match[4]}:#{match[5]}:#{match[6]}.#{format('%06d',
+                                                                                                   microsec)}+00:00"
```

Comment:
Or it would be even nicer to have meaningful descriptions for each part of this string, like:

```ruby
day = "#{match[1]}-#{match[2]}-#{match[3]}"
time_with_seconds = "#{match[4]}:#{match[5]}:#{match[6]}"
microseconds = format('%06d', microsec)
date = "#{day}T#{time_with_seconds}.#{microseconds}+00:00"
```